images/bootstrap/runner.sh

```diff
@@ -41,94 +41,64 @@ setup_ca(){
     fi
 }
 
 early_exit_handler() {
-    cleanup_dind
+    cleanup_pinc
 }
 
 # setup certificates before anything gets started
 setup_ca
 
-# optionally enable ipv6 docker
-export DOCKER_IN_DOCKER_IPV6_ENABLED=${DOCKER_IN_DOCKER_IPV6_ENABLED:-false}
-if [[ "${DOCKER_IN_DOCKER_IPV6_ENABLED}" == "true" ]]; then
-    echo "Enabling IPV6 for Docker."
-    # configure the daemon with ipv6
-    mkdir -p /etc/docker/
-    cat <<EOF >/etc/docker/daemon.json
-{
-  "ipv6": true,
-  "fixed-cidr-v6": "fc00:db8:1::/64"
-}
-EOF
+# optionally enable ipv6
+export PODMAN_IN_CONTAINER_IPV6_ENABLED=${PODMAN_IN_CONTAINER_IPV6_ENABLED:-false}
+if [[ "${PODMAN_IN_CONTAINER_IPV6_ENABLED}" == "true" ]]; then
+    echo "Enabling IPV6."
     # enable ipv6
     sysctl net.ipv6.conf.all.disable_ipv6=0
     sysctl net.ipv6.conf.all.forwarding=1
     # enable ipv6 iptables
     modprobe -v ip6table_nat
 fi
 
-# Check if the job has opted-in to docker-in-docker availability.
-export DOCKER_IN_DOCKER_ENABLED=${DOCKER_IN_DOCKER_ENABLED:-false}
-if [[ "${DOCKER_IN_DOCKER_ENABLED}" == "true" ]]; then
-    echo "Docker in Docker enabled, initializing..."
-
-    export DOCKER_DEBUG=${DOCKER_DEBUG:-false}
-    if [[ "${DOCKER_DEBUG}" == "true" ]]; then
-        mkdir -p /etc/docker/
-        # TODO: do not rely on this file not existing!
-        cat <<EOF >/etc/docker/daemon.json
-{
-  "debug": true,
-  "log-level": "debug"
-}
-EOF
-    fi
-
-    printf '=%.0s' {1..80}; echo
-    # If we have opted in to docker in docker, start the docker daemon,
+export PODMAN_IN_CONTAINER_ENABLED=${PODMAN_IN_CONTAINER_ENABLED:-false}
+if [[ "${PODMAN_IN_CONTAINER_ENABLED}" == "true" ]]; then
+    echo "Podman in Container enabled, initializing in podman compatible mode..."
     (
-        if [ -f "/etc/default/docker" ]; then
-            source /etc/default/docker
-        fi
-        /usr/bin/dockerd \
-            -p /var/run/docker.pid \
-            --data-root=/docker-graph \
-            --init-path /usr/libexec/docker/docker-init \
-            --userland-proxy-path /usr/libexec/docker/docker-proxy \
-            ${DOCKER_OPTS} \
-                >/var/log/dockerd.log 2>&1 &
+        export HTTP_PROXY=${CONTAINER_HTTP_PROXY}
+        export HTTPS_PROXY=${CONTAINER_HTTPS_PROXY}
+	mkdir -p ${XDG_RUNTIME_DIR}/podman
+        podman system service \
+               -t 0 \
+               unix:///${XDG_RUNTIME_DIR}/podman/podman.sock \
+               >/var/log/podman.log 2>&1 &
+        echo "${!}" > /var/run/podman.pid
     )
-    # the service can be started but the docker socket not ready, wait for ready
+    # the service can be started but the socket not ready, wait for ready
     WAIT_N=0
     MAX_WAIT=5
     while true; do
-        # docker ps -q should only work if the daemon is ready
-        docker ps -q > /dev/null 2>&1 && break
+        # podman ps -q should only work if the daemon is ready
+        podman ps -q > /dev/null 2>&1 && break
```

Comment:
Maybe it would be better to use the socket for testing if podman works? Like done here: https://github.com/kubevirt/kubevirtci/blob/23ddfc29c03bb1ccaeb79cd204163f8108db1c36/cluster-up/cluster/ephemeral-provider-common.sh#L24

Reply:
Or here https://github.com/kubevirt/kubevirtci/blob/23ddfc29c03bb1ccaeb79cd204163f8108db1c36/cluster-up/cluster/ephemeral-provider-common.sh#L18

Reply:
yes good points - I will look into improving this check - thanks @0xFelix !

Reply:
yes good points - I will look into improving this check - thanks @0xFelix !